smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/generators/GetterGenerator.java

```diff
@@ -136,12 +137,16 @@ public Void structureShape(StructureShape shape) {
                     // If the member targets a collection shape and is optional then generate an unwrapped
                     // getter as a convenience method as well.
                     Shape target = model.expectShape(member.getTarget());
-                    if (target.isListShape() || target.isMapShape()) {
+                    boolean isListShape = target.isListShape();
+                    if (isListShape || target.isMapShape()) {
                         writer.openBlock("public $T get$UOrEmpty() {",
                                 "}",
                                 symbolProvider.toSymbol(member),
                                 symbolProvider.toMemberName(member),
-                                () -> writer.write("return $L;", symbolProvider.toMemberName(member)));
+                                () -> writer.write("return $1L == null ? $2T.empty$3L() : $1L;",
+                                        symbolProvider.toMemberName(member),
+                                        Collections.class,
+                                        isListShape ? "List" : "Map"));
```

Comment:
Could you do just `Collections.emptyList()`/`Collections.emptyMap()` instead of having to interpolate? It would be a tad clearer, but no big deal.

Reply:
I guess we can, but we need to branch somewhere. What about spelling `emptyList` and `emptyMap` instead of just `List` and `Map`. That might be clearer.